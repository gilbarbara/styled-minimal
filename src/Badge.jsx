import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { px, themeGet } from './utils/helpers';
import { baseStyles, sizesAllPropTypes, variantPropTypes } from './utils/system';

import Box from './Box';

export const Badge = styled(Box)`
  ${(props: Object): string => {
    const {
      borderRadius,
      fontSize,
      fontWeight,
      padding,
    } = themeGet(props, 'badge');
    const fontSizeProp = themeGet(props, 'componentSizes', { key: 'size' });

    return css`
      ${baseStyles.variant};
      border-radius: ${px(borderRadius)};
      display: inline-flex;
      font-size: ${fontSize};
      ${fontSizeProp ? `font-size: ${fontSizeProp}` : ''};
      font-weight: ${fontWeight};
      line-height: 1;
      padding: ${px(padding[0])} ${px(padding[1])};
      vertical-align: baseline;
    `;
  }};
`;

Badge.displayName = 'Badge';

Badge.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  dark: PropTypes.bool,
  outline: PropTypes.bool,
  size: sizesAllPropTypes,
  variant: variantPropTypes,
  ...Box.propTypes,
};

Badge.defaultProps = {
  as: 'span',
};

export default Badge;
