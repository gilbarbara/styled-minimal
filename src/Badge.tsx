import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import { baseStyles, sx, system, variantStyles } from './utils/system';

import { BadgeProps } from './types';

export const StyledBadge = styled('sup')<BadgeProps>(
  props => {
    const { dot } = props;
    const { borderRadius, fontSize, fontWeight, lineHeight, padding, size, sizeDot } = getTheme(
      props,
      'badge',
    );

    return css`
      ${baseStyles};
      ${variantStyles};
      border-radius: ${dot ? '100%' : px(borderRadius)};
      display: inline-flex;
      font-size: ${dot ? 0 : px(fontSize)};
      font-weight: ${fontWeight};
      justify-content: center;
      line-height: ${dot ? 0 : px(lineHeight)};
      height: ${px(dot ? sizeDot : size)};
      padding: ${dot ? 0 : `0 ${px(padding)}`};
      min-width: ${px(dot ? sizeDot : size)};
    `;
  },
  sx,
  system,
);

export const Badge = React.forwardRef<HTMLSpanElement, BadgeProps>((props, ref) => (
  <StyledBadge ref={ref} {...props} />
));

Badge.defaultProps = {
  dark: false,
  dot: false,
  invert: false,
  variant: 'primary',
};
