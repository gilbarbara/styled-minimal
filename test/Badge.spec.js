import React from 'react';
import { mount } from 'enzyme';

import Badge from '../src/Badge';

describe('Badge', () => {
  const wrapper = mount(
    <Badge>4</Badge>
  );

  it('should ', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
