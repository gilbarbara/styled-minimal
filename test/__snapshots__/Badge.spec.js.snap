// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should  1`] = `
.c0 {
  box-sizing: border-box;
  background-color: #2E2E2E;
  border: 1px solid #2E2E2E;
  color: #fff;
  border-radius: 3px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  padding: 3px 6px;
  vertical-align: baseline;
}

<Badge
  as="span"
>
  <span
    className="c0"
  >
    4
  </span>
</Badge>
`;
